### PR TITLE
Added a crawler to extract NtStatus definitions from MSDN

### PR DESCRIPTION
**Pull request by oec**
Hi,

I wrote a simple crawler based on github.com/PuerkitoBio/goquery to extract the NtStatus codes from https://msdn.microsoft.com/en-us/library/cc704588.aspx. In ntstatus.go, a go:generate now calls statuscrawler which generates ntstatus_generated.go.

ntstatus_generated.go contains the consts as well as the String() method for NtStatus to derive the STATUS_* name for a given NtStatus (via a map[NtStatus]string{}.  The stringer-tool, for some reason, did not provide a properly generated go-file).

Hope this is what you had in mind with your "FIXME"-comment in ntstatus.go.

Cheers,
Özgür

PS: I also fixed a bug that did not allow the code to compile, there were casts missing to uint32 from the last commit.

### CODE REVIEW

**Comment by hillu** [comment]
Woohoo, awesome, thank you. I am going to have a closer look within the next few days.

**Comment by oec** [comment]
Please do.  If you already have a corpus of test cases to run against, that would be helpful.

File: examples/lsobj/lsobj.go [hunk 1]
```diff
@@ -5,10 +5,11 @@ package main
 import (
 	"errors"
 	"fmt"
+	"log"
 	"os"
 	"unsafe"
 
-	"github.com/hillu/go-ntdll"
+	"github.com/oec/go-ntdll"
```
Comment (rasa) [inline examples/lsobj/lsobj.go]: Need to flip this back to hillu, eventually

Comment (oec) [reply examples/lsobj/lsobj.go]: yeah, let me flip that back. needed it to test.

Comment (oec) [reply examples/lsobj/lsobj.go]: reverted and checked in a new patch

**Comment by oec** [comment]
@rasa , I picked up on your great idea of sorting the status codes and extended it a bit:  I sort the original list of status codes directly and can no print both, the list of consts and the map in order of status code.

**Comment by rasa** [comment]
@oec Yeah, that's much better, cool.

**Comment by oec** [comment]
@hillu did you have chance to look into this PR?  I have another one open and plan to add at least one more Nt* function (NtDeviceIoControlFile) during the next couple of weeks.

**Comment by hillu** [comment]
@oec I like the change in general, but there are some things you should change:
- There should be just one change for the crawling script
- The generated code should not contain the timestamp based on `time.Now()`. If you want to add a timestamp, it should be based on the last change of the MSDN page.

**Comment by hillu** [comment]
@oec It sounds like a change regarding other functions is independent of this change. Please put it into a separate PR.

**Comment by oec** [comment]
@hillu sounds fair.  So I will

1. cancel the PR
2. create one PR for the missing uint32 cast
3. change the crawler to print a timestamp from the MSDN page (probably the value from the `MN`-meta-tag) and create a PR for it
4. create one PR for adding NtDeviceIoControlFile

**Comment by hillu** [comment]
Just git push --force to the branch in your fork and reopen this PR when you are ready.

**Comment by oec** [comment]
Can you just ignore this PR and check&merge the other one I opened already?  It is a combined patch that also contains a different timestamp.

**Pull request closed without merging**